Source_Code/TM_UnitTests/TeamMentor.UnitTests/TeamMentor.UserData/Users/Test_Users_Creation.cs

```diff
@@ -408,6 +467,76 @@ public void CreateTmUserSigupResponse_Email_AlreadyExist()
+            validations.FirstOrDefault().Message.assert_Is(TMConsts.DEFAULT_EMAIL_ADDRESS_IS_INVALID);
+        }
+        [Test]
+        public void CreateTmUserSigupResponse_Email_Address_Large()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+            var newUser = new NewUser().with_Random_Data();
+            newUser.Email = "".random_Email().add_RandomLetters(300);
+            
+            var response = userData.createTmUserResponse(newUser);
+            response.assert_Is_Not_Null();
+            response.UserCreated.assert_Is(0);
+            response.Signup_Status.assert_Is(Signup_Result.SignupStatus.Validation_Failed);
+            response.Validation_Results.assert_Not_Null();
+            response.Validation_Results.count().assert_Bigger_Than(0);
+            response.Validation_Results.FirstOrDefault().Field.assert_Is("Email");
+            response.Validation_Results.FirstOrDefault().Message.assert_Is(TMConsts.DEFAULT_EMAIL_ADDRESS_IS_INVALID);
+         }
+
+        [Test]
+        public void CreateTmUserSigupResponse_Email_Address_IsNull()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+            var newUser = new NewUser().with_Random_Data();
+            newUser.Email = null;
+            var response = userData.createTmUserResponse(newUser);
+            response.assert_Not_Null();
+            response.UserCreated.assert_Is(0);
+            response.Signup_Status.assert_Is(Signup_Result.SignupStatus.Validation_Failed);
+            response.Validation_Results.count().assert_Bigger_Than(0);
+            response.Validation_Results.FirstOrDefault().Field.assert_Is("Email");
+            response.Validation_Results.FirstOrDefault().Message.assert_Is(TMConsts.DEFAULT_EMAIL_ADDRESS_IS_INVALID);
+
+        }
+       
+        [Test]
+        public void CreateTmUserSigupResponse_Email_Address_NullorEmpty()
+        {
+            UserGroup.None.assert();                //  change current thread privildges to None
+            var newUser = new NewUser()
+            {
+                Username = "tA2!".add_RandomLetters(8),
+                Email = String.Empty
+            };
+            var response = userData.createTmUserResponse(newUser);
+            Assert.IsTrue((response != null));
+            Assert.IsTrue(response != null && response.UserCreated == 0);
+            Assert.IsTrue(response != null && response.Signup_Status == Signup_Result.SignupStatus.Validation_Failed);
+            if (response != null)
+            {
+                var results = response.Validation_Results;
+                Assert.IsTrue(results.count() > 0);
+                Assert.IsTrue(results.FirstOrDefault().Field == "Email");
+                Assert.IsTrue(results.FirstOrDefault().Message == TMConsts.DEFAULT_EMAIL_ADDRESS_IS_INVALID);
+            }
+        }

```

Comment:
Weird, I changed that code you can actually see it in this other visualization

https://github.com/TeamMentor/Dev/pull/173/files#diff-8b5f047f12caf1636db14063fff24709R524

I think I will close this Pull Request and open up a new one because this one is kind of messy.

What do you think?